aws-storage-s3/src/test/java/com/amplifyframework/storage/s3/StorageComponentTest.java

```diff
@@ -264,6 +266,42 @@ public void testUploadFileGetsKey() throws Exception {
         assertEquals(toRemoteKey, result.getKey());
     }
 
+    /**
+     * Test that calling upload inputStream method from Storage category correctly
+     * invokes the registered AWSS3StoragePlugin instance and returns a
+     * {@link StorageUploadFileResult} with correct remote key.
+     *
+     * @throws Exception when an error is encountered while uploading
+     */
+    @Test
+    public void testUploadInputStreamGetsKey() throws Exception {
+        final String toRemoteKey = RandomString.string();
+        final InputStream inputStream = new ByteArrayInputStream("test data".getBytes());
```

Comment:
You could create a `RandomBytes` (like the `RandomString` utility) under testutils?

```java
new ByteArrayInputStream(RandomBytes.bytes());
```
with
```java
public final class RandomBytes {
    private static final long DEFAULT_BYTE_COUNT = // choose one;
    private RandomBytes() {}

    public static byte[] bytes() {
        return bytes(DEFAULT_BYTE_COUNT);
    }

    public static byte[] bytes(int count) {
        // ...
    }
}
```